core/src/main/java/org/mskcc/cbio/oncokb/util/EvidenceUtils.java

```diff
@@ -953,11 +953,35 @@ private static Gene getGene(Integer entrezGeneId, String hugoSymbol) {
     }
 
     public static void annotateEvidence(Evidence evidence) throws ParserConfigurationException {
-        ClinicalTrialBo clinicalTrialBo = ApplicationContextSingleton.getClinicalTrialBo();
-        ArticleBo articleBo = ApplicationContextSingleton.getArticleBo();
-        NccnGuidelineBo nccnGuidelineBo = ApplicationContextSingleton.getNccnGuidelineBo();
-        DrugBo drugBo = ApplicationContextSingleton.getDrugBo();
-        TreatmentBo treatmentBo = ApplicationContextSingleton.getTreatmentBo();
+        GeneBo geneBo = ApplicationContextSingleton.getGeneBo();
+        Gene gene = geneBo.findGeneByHugoSymbol(evidence.getGene().getHugoSymbol());
+        evidence.setGene(gene);
```

Comment:
In the updateEvidenceBasedOnUuid from Evidence Controller, I check if gene is null. Thus we do need to set gene in annotateEvidence even if it is null.